### DISCUSSION TITLE
LGPL instead of AGPL

### DISCUSSION THREAD

**Discussion by CaptainPaule** [weiznich/diesel_async#5] [Q&A] [answered]
Hello,

first I want to say thank you for this great project! I am looking for a good implementation to use diesel in a async way for a while now, but unfortunately the current AGPLv3 licence is not an option in my current long term project. There was a remark about reaching out for a more permissive licence. I want to ask for a possibility to use diesel_asnyc with LGPLv3 to be able to link against without useing
AGPLv3 for the complete project.

In my opinion it would be nicer to use this project as dependency and contribute here in case of a sensefull feature instead of do my own implementation.

thank you for help!
Paul

**Comment by weiznich** [comment, marked as answer]
Thanks for reaching out here. 
I'm currently not interested to change the license to something more permissive as this would likely lead to more people using `diesel_async` and I'm not sure if I really can commit that much time on it the next few month. At least I personally do not  need `diesel_async` for my projects. It currently exists as a demonstration what is possible.

That does not mean that I'm fundamentally against changing the license someday. It's just that I do not have the resources to maintain it correctly for a foreseeable future. This could be changed by either:

* More people showing up to help maintaining the diesel ecosystem at all. As it currently stands there are too few people helping out. Everyone seems to just expect to take anything for free. This seems to be especially true for the async ecosystem where basically all relevant use cases involve a professional usage. Therefore the current restrictive license.
* Someone starts to pay me for actively maintaining diesel and the surrounding ecosystem.

If you or your company want to support me to actively maintain the Diesel ecosystem please reach out to diesel@weiznich.de

> In my opinion it would be nicer to use this project as dependency and contribute here in case of a sensefull feature instead of do my own implementation.

If you choose to develop your own implementation, please keep in mind that you cannot use this crate as source for inspirations as that would be derived work, which falls under the same license as the crate itself. This likely includes parts of the API as there are few parts in there that are non-trivial.
